clients/src/main/java/org/apache/kafka/clients/admin/internals/CoordinatorStrategy.java

```diff
@@ -98,7 +98,8 @@ public ApiRequestScope lookupScope(CoordinatorKey key) {
                 key.idValue + "' cannot be represented in a request."));
         }
         FindCoordinatorResponse response = (FindCoordinatorResponse) abstractResponse;
-        if (batch) {
+        // Handle response based on whether batched version is used by checking `coordinators()` in the response data
+        if (!response.data().coordinators().isEmpty()) {
```

Comment:
Both of consumer coordinator and transaction coordinator parse response according to version. By contrast, it check `coordinators`. Could we unify the behavior? Could consumer coordinator and transaction coordinator check `coordinators` instead? It seems checking `coordinators` can simplify the code as we don't need to pass request version to each handler.